File: emailtemplate/src/routes/meta.ts
import { Response } from "express";
import { AppRequest } from "../types/appRequest";
import { getMetaFromFactors } from "../utils/meta";
import { Meta } from "../types/meta";

type MetaResponse = {
  status: "ok" | "error",
  error: string | null,
  data: Meta,
};

export async function metaRoute(req: AppRequest, res: Response) {
  const meta = getMetaFromFactors(req.appContext.metaFactors);

  return res.status(200).json({
    status: "ok",
    error: null,
    data: meta,
  } satisfies MetaResponse);
}

File: module/routes/meta.go
package routes

import (
	"github.com/autobar-dev/services/module/types"
	"github.com/autobar-dev/services/module/utils"

	"github.com/labstack/echo/v4"
)

type MetaRouteResponse struct {
	Status string      `json:"status"`
	Data   *types.Meta `json:"data"`
}

func MetaRoute(c echo.Context) error {
	rest_context := c.(*types.RestContext)
	app_context := *(*rest_context).AppContext

	meta := utils.GetMetaFromFactors(app_context.MetaFactors)

	return rest_context.JSON(200, &MetaRouteResponse{
		Status: "ok",
		Data:   meta,
	})
}
